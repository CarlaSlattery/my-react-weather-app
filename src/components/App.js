import React, { useEffect, useState } from "react";
import getForecast from "../requests/getForecast";

import LocationDetails from "./LocationDetails";
import ForecastSummaries from "./ForecastSummaries";
import ForecastDetails from "./ForecastDetails";
import SearchForm from "./SearchForm";

import "../styles/App.css";
import "../styles/ForecastSummaries.css";

function App() {
  const [forecasts, setForecasts] = useState([]);
  const [searchText, setSearchText] = useState("");
  const [location, setLocation] = useState({ city: "", country: "" });
  const [selectedDate, setSelectedDate] = useState(0);
  const [errorMessage, setErrorMessage] = useState("");

  const handleCitySearch = () => {
    getForecast(
      searchText,
      setSelectedDate,
      setForecasts,
      setLocation,
      setErrorMessage
    );
  };
  const selectedForecast = forecasts.find(
    (forecast) => forecast.date === selectedDate
  );
  const handleForecastSelect = (date) => {
    setSelectedDate(date);
  };

  useEffect(() => {
    getForecast(setSelectedDate, setForecasts, setLocation);
  }, []);

  return (
    <div className="weather-app">
      <LocationDetails
        city={location.city}
        country={location.country}
        errorMessage={errorMessage}
      />
      <SearchForm
        searchText={searchText}
        setSearchText={setSearchText}
        onSubmit={handleCitySearch}
      />
      {!errorMessage && (
        <>
          <ForecastSummaries
            forecasts={forecasts}
            onForecastSelect={handleForecastSelect}
          />
          {selectedForecast && <ForecastDetails forecast={selectedForecast} />}
        </>
      )}
    </div>
  );
}

export default App;
