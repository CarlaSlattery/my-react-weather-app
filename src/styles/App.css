body {
  background-color: rgb(147, 178, 203);
}

.weather-app {
  margin: 0 20px;
}
