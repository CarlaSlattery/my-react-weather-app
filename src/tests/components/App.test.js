import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import App from "../../components/App";

describe("App", () => {
  it("renders app component correctly", () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders and errorMessage if API call not recognised", async () => {
    render(<App />);

    const searchFormInput = await screen.findByTestId("search-form-input");
    fireEvent.change(searchFormInput, { target: { value: "Paris" } });

    expect(
      await screen.findByText("Location not recognised - please try again")
    );
  });
});
