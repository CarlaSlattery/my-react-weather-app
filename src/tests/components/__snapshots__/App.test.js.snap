// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders app component correctly 1`] = `
<DocumentFragment>
  <div
    class="app-container"
  >
    <h1>
      Five Day Weather Forecast
    </h1>
    <h2>
      , 
    </h2>
    <div
      class="search-form"
    >
      <input
        data-testid="search-form-input"
        placeholder="Enter Location"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Search
      </button>
    </div>
    <div
      class="forecast-summaries"
    />
  </div>
</DocumentFragment>
`;
